### DISCUSSION TITLE
Custom audio player integration using this script

### DISCUSSION THREAD

**Discussion by abuhasan0319** [katspaugh/wavesurfer.js#3491] [Q&A]
Hi, I am using this script to customize the integration of an audio player so that I can add functionality based on my requirements. However, when I use this script, it takes too long to load and appears on the UI. Could you please look into it and suggest a better way of doing it?
`
var wavesurfer = WaveSurfer.create({
    container: "#audiowave",
    waveColor: "rgb(200, 0, 200)",
    progressColor: "rgb(100, 0, 100)",
    url: 'test.url',
    height: 100,
    responsive: true,
    hideScrollbar: true,
    cursorWidth: 2,
    barWidth: 2,
    barGap: 2,
    skipLength: 5,
    mediaControls: true,
});

const playButton = document.querySelector('#play')
const forwardButton = document.querySelector('#forward')
const backButton = document.querySelector('#backward')

wavesurfer.once('decode', () => {
    document.querySelectorAll('input[type="checkbox"]').forEach((input) => {
        input.onchange = (e) => {
            wavesurfer.setOptions({
                [input.value]: e.target.checked,
            })
        }
    })

    playButton.onclick = () => {
        wavesurfer.playPause()
    }

    forwardButton.onclick = () => {
        wavesurfer.skip(5)
    }

    backButton.onclick = () => {
        wavesurfer.skip(-5)
    }
})

`

**Comment by katspaugh** [comment]
Your code looks alright. It may take long to load if your audio file is big or the server is slow.

**Reply by abuhasan0319** [reply]
Hi, I am using a secured audio URL, but when I use it with wavesufer.js, it takes time to appear on the UI. However, when I am not using wavesufer.js and simply open the URL, it does not take any time at all to load the URL in the audio tag. Please assist me with this issue.

**Reply by katspaugh** [reply]
Please don’t create additional threads. You can find some solutions to slow loading in the FAQ in the readme.

**Reply by abuhasan0319** [reply]
I didn't found any solution in FAQ, please suggest me better solution. The size of my audio file is 12 MB and length of audio is 5 min.

**Reply by abuhasan0319** [reply]
Hi, Its been 1 week you haven't replied yet?
